src/frontend/src/pages/AdminToolsPage/ProjectsConfig/LinkTypeFormModal.tsx

```diff
@@ -0,0 +1,120 @@
+import { Controller, useForm } from 'react-hook-form';
+import NERFormModal from '../../../components/NERFormModal';
+import { FormControl, FormLabel, FormHelperText, Switch, Stack, Box, Typography, Tooltip, Link } from '@mui/material';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+import { useToast } from '../../../hooks/toasts.hooks';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { LinkType } from 'shared';
+import { LinkTypeCreatePayload } from '../../../utils/types';
+import Icon from '@mui/material/Icon';
+import HelpIcon from '@mui/icons-material/Help';
+
+interface LinkTypeFormModalProps {
+  showModal: boolean;
+  handleClose: () => void;
+  defaultValues?: LinkType;
+  onSubmit: (data: LinkTypeCreatePayload) => void;
+  linkTypes: LinkType[];
+}
+
+const LinkTypeFormModal = ({ showModal, handleClose, defaultValues, onSubmit, linkTypes }: LinkTypeFormModalProps) => {
+  const toast = useToast();
+
+  const uniqueLinkTypeTest = (name?: string) => !!name && linkTypes && !linkTypes.map((v) => v.name).includes(name);
+
+  const schema = yup.object().shape({
+    name: yup
+      .string()
+      .required('LinkType Name is Required')
+      .test('unique-LinkType-test', 'LinkType name must be unique', uniqueLinkTypeTest),
+    iconName: yup.string().required('Icon name is required'),
+    required: yup.boolean().required('Required field must be specified')
+  });
+
+  const {
+    handleSubmit,
+    control,
+    reset,
+    formState: { errors },
+    watch
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      name: defaultValues?.name ?? '',
+      iconName: defaultValues?.iconName ?? '',
+      required: defaultValues?.required ?? false
+    }
+  });
+
+  const iconNameWatch = watch('iconName');
+
+  const onFormSubmit = async (data: LinkTypeCreatePayload) => {
+    try {
+      await onSubmit(data);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        toast.error(error.message);
+      }
+    }
+    handleClose();
+  };
+  const tooltipMessage = (
+    <Typography sx={{ fontSize: 14 }}>
+      Click to view possible icon names. For names with multiple words, seperate them with an _. AttachMoney = attach_money
+    </Typography>
+  );
+  return (
+    <NERFormModal
+      open={showModal}
+      onHide={handleClose}
+      title={!!defaultValues ? 'Edit LinkType' : 'Create LinkType'}
+      reset={() => reset({ name: '' })}
+      handleUseFormSubmit={handleSubmit}
+      onFormSubmit={onFormSubmit}
+      formId={!!defaultValues ? 'edit-LinkType-form' : 'create-LinkType-form'}
+      showCloseButton
+    >
+      <Stack spacing={2} sx={{ width: '100%' }}>
```

Comment:
Let's use the space we have, so do Name and Required on the top row and then Icon Name with the preview to the right of it on the second row